_posts/2024-09-12-v5-release.md

```diff
@@ -0,0 +1,139 @@
+
+#### 2. Splats, optional, and captures oh my. Simplified patterns for common route patterns.
+
+With the removal of regular expression semantics comes other small but impactful changes to how you write your routes.
+
+1. `:name?` becomes `{:name}`. Usage of `{}` for optional parts of your route means you can now do things like `/base{/:optional}/:required` and what parts are actually optional is
+   much more explicit.
+2. `:name*` becomes `*name`. (@blakeembrey to provide more details)
+3. `:name+` is equivalent to a `*name` and so has been removed
+4. New reserved characters: `(`, `)`, `[`, `]`, `?`, `+`, & `!`. These have been reserved to leave room for future improvements and to prevent mistakes when migrating where those
+   characters mean specific things in previous versions.
+
+*How to migrate:* (@wesleytodd to provide more info with codemods and migration tools) 
+
+#### 3. Name everything. Ordered numerical parameters are not supported.
+
+In Express v4, using regex capture groups you could get numerical parameters (ex. `/user(s?)` => `req.params[0] === 's'`). Now all params must be named. Along with requiring a
+name, we now support all valid JavaScript identifiers or quoted (ex. `/:"this"`).
+
+### Promise Support
+
+This one may be a bit contentious, but I `Promise` we are moving in the right direction. We have added support for returned *rejected* promises from middleware raising errors. This
+*does not include* calling `next` from returned *resolved* promises. There are a lot of edge cases in old express apps around expectations of `Promise` behavior, and before we can
+run we need to walk. For most folks, this means you can now write middleware like the following:
+
+```javascript
+app.use(async (req, res, next) => {
+  req.locals.user = await getUser(req);
+  next();
+});
+```
+
+Notice that we use `async/await` and the `getUser` call may throw (user does not exist, db is down, etc), but we still call `next` if it is successful. We dont need to catch the
+error in line anymore if we want to rely on error handling middleware instead because the router will now catch the rejected promise and treat that as calling `next(err)`.
+
+*editorial note:* Error handling is a huge topic, but one hill I will die on is that errors should be handled as close to the error site as possible. So while this is now handled
+in the router, I would strongly urge you to catch this kind of error within the middleware and handle it without relying on separate error handling middleware.
+
+### Body Parser
+
+We made an assortment of `body-parser` changes:
+
+- Add option to customize the urlencoded body depth with a default value of 32 (@TODO see CVE)
+- Remove deprecated `bodyParser()` combination middleware
+- `req.body` is no longer always initialized to `{}`
+- `urlencoded` parser now defaults `extended` to false
+- Added brotli support
+
+### Removed APIs
```

Comment:
Since its a blog and probably the interested ones will read it, would it be a good idea to include a reason or link to something so that users can understand why the removed APIs were removed?

Reply:
I like the idea but I think that will make the blog post longer to read. Also the links probably will be only links to commits so it will be hard to contextualize, maybe we can include a sentence to summarize the strategy that we follow as a framework.

Reply:
I added an explanation in eea125540482980f218c67a8a8cc355c15120cba